packages/clerk-js/src/ui/components/SessionTasks/index.tsx

```diff
@@ -93,8 +102,19 @@ export const SessionTasks = withCardStateProvider(({ redirectUrlComplete }: Sess
       return;
     }
 
+    // If the current path does does not match any of the tasks, navigate away to the initial path,
+    // this handles cases where a user uses browser back navigation to the tasks URL.
+    if (
+      task &&
+      clerk.session?.tasks &&
+      clerk.session.tasks.length > 0 &&
+      !clerk.session.tasks.find(t => matches(INTERNAL_SESSION_TASK_ROUTE_BY_KEY[t.key]))
```

Comment:
`INTERNAL_SESSION_TASK_ROUTE_BY_KEY[t.key])` returns the UI route name for the task, not the task key itself that matches FAPI, so if I'd rename the value of `reset-password` within that map, this effect would break. 

Is the intent here to navigate from `/tasks` to `/tasks/reset-password` or `/tasks/choose-organization` to `/tasks/reset-password`?